### PR TITLE
implement blocking filtering

### PR DESCRIPTION
**Pull request by akmatchev**
## Overview

Filtered posts by blocked users. 
NOTE: Works in postman, but test suite fails, will debug later.

## Next Steps (delete if not applicable)

Debug test suite

## Screenshots (delete if not applicable)

<details>

  <summary>Screen Shot Name</summary>

  
  

</details>

### CODE REVIEW

**Review by kidzegeye — APPROVED** [review]
Looks good

File: src/services/PostService.ts [hunk 1]
```diff
@@ -21,30 +21,56 @@ export class PostService {
...
+      const blockedUsers = userWithBlockedInfo?.blocking;
+      const unblockedPosts = activeUserPosts.filter((post) => {
+        if (blockedUsers) {
+          for (const blockedUser of blockedUsers) {
+            if (post.user?.id == blockedUser.id) return false;
+          }
+        }
+        return true;
+      });
+      return unblockedPosts;
```
Comment (Archit404Error) [inline src/services/PostService.ts:39]: It looks like this code is duplicated in some of the other functions, it'd be a good idea to factor this out into a separate helper method called something like `filterBlockedUserPosts()`, which takes in a set of posts and a user id and then returns a filtered version of the posts. You can then just take whatever posts a given route was originally and pass them through this function -- i.e., if a service was originally returning `posts` which you fetched from the DB, now just return `filterBlockedUserPosts(posts, user.id)`. This should make your code easier to change in the future and avoid the duplication

File: src/services/PostService.ts [hunk 2]
```diff
@@ -21,30 +21,56 @@ export class PostService {
...
       if (!post) throw new NotFoundError('Post not found!');
       if (!post.user?.isActive) throw new NotFoundError('User is not active!');
+      const userRepository = Repositories.user(transactionalEntityManager);
+      const userWithBlockedInfo = await userRepository.getUserWithBlockedInfo(user.id)
+      const blockedUsers = userWithBlockedInfo?.blocking;
+      if (blockedUsers) {
+        for (const blockedUser of blockedUsers) {
+          if (post.user?.id == blockedUser.id) throw new ForbiddenError('User is blocked!');
+        }
+      }
```
Comment (Archit404Error) [inline src/services/PostService.ts:56]: Same comment here as the one above, you can write an additional helper method that just throws an error whenever a post or array of posts contains a post from someone a user has blocked called something like `validatePostNotBlocked()`, and then just call this function everywhere you have this logic

File: src/services/PostService.ts [hunk 3]
```diff
@@ -21,30 +21,56 @@ export class PostService {
...
       const user = await userRepository.getUserById(params.id)
       if (!user) throw new NotFoundError('User not found!');
       if (!user.isActive) throw new NotFoundError('User is not active!');
+      const userWithBlockedInfo = await userRepository.getUserWithBlockedInfo(currentUser.id)
+      const blockedUsers = userWithBlockedInfo?.blocking;
+      if (blockedUsers) {
+        for (const blockedUser of blockedUsers) {
+          if (user.id == blockedUser.id) throw new ForbiddenError('User is blocked!');
+        }
+      }
```
Comment (Archit404Error) [inline src/services/PostService.ts:73]: Replace with helper method call

File: src/services/PostService.ts [hunk 4]
```diff
@@ -116,30 +142,78 @@ export class PostService {
...
+      const blockedUsers = userWithBlockedInfo?.blocking;
+      const unblockedPosts = filteredPosts.filter((post) => {
+        if (blockedUsers) {
+          for (const blockedUser of blockedUsers) {
+            if (post.user?.id == blockedUser.id) return false;
+          }
+        }
+        return true;
+      });
+      return unblockedPosts;
```
Comment (Archit404Error) [inline src/services/PostService.ts:157]: replace with helper method call

File: src/services/PostService.ts [hunk 5]
```diff
@@ -116,30 +142,78 @@ export class PostService {
...
```
Comment (Archit404Error) [inline src/services/PostService.ts:177]: replace with helper method call

File: src/services/PostService.ts [hunk 6]
```diff
@@ -116,30 +142,78 @@ export class PostService {
...
```
Comment (Archit404Error) [inline src/services/PostService.ts:197]: same comment

File: src/services/PostService.ts [hunk 7]
```diff
@@ -116,30 +142,78 @@ export class PostService {
...
```
Comment (Archit404Error) [inline src/services/PostService.ts:216]: same comment

File: src/services/PostService.ts [hunk 8]
```diff
@@ -256,7 +330,19 @@ export class PostService {
...
```
Comment (Archit404Error) [inline src/services/PostService.ts:345]: same comment

**Review by Archit404Error — APPROVED** [review]
Great work on this PR Tony! The overall logic looks good and makes sense, I'd suggest just factoring out a lot of the duplicate code in the services into 2 helper methods that you can just call to either filter posts or validate that a post is not from a blocked user. 

This will help make the code more readable and maintainable!

**Pull request merged**